app/scripts/metamask-controller.js

```diff
@@ -436,28 +436,24 @@ module.exports = class MetamaskController extends EventEmitter {
    * @returns {Object} vault
    */
   async createNewVaultAndKeychain (password) {
-    const release = await this.createVaultMutex.acquire()
-    let vault
-
+    const releaseLock = await this.createVaultMutex.acquire()
```

Comment:
Can the mutex acquisition ever fail? Should we move this asynchronous operation inside the try / catch as well?

Reply:
acquisition should never fail